pkg/config/initoptions.go

```diff
@@ -0,0 +1,179 @@
+type InitOptions struct {
+	ioStreams genericclioptions.IOStreams
+	// Package directory argument
+	Dir string
+	// Namespace for grouping object
+	Namespace string
+	// Grouping object label value
+	GroupName string
+	// Random seed
+	Seed int64
+}
+
+func NewInitOptions(ioStreams genericclioptions.IOStreams) *InitOptions {
+	return &InitOptions{
+		ioStreams: ioStreams,
+	}
+}
+
+// Complete fills in the InitOptions fields.
+func (i *InitOptions) Complete(args []string) error {
+	i.Seed = time.Now().UnixNano()
+	if len(args) != 1 {
+		return fmt.Errorf("need one 'directory' arg; have %d", len(args))
+	}
+	i.Dir = args[0]
+	if !isDirectory(i.Dir) {
+		return fmt.Errorf("invalid directory argument: %s", i.Dir)
+	}
+	if len(i.Namespace) == 0 {
+		namespace, err := packageNamespace(i.Dir)
+		if err != nil {
+			return err
+		}
+		i.Namespace = namespace
+	}
+	if len(i.GroupName) == 0 {
+		i.GroupName = i.defaultGroupName()
+	}
+	if !validateGroupName(i.GroupName) {
+		return fmt.Errorf("invalid group name: %s", i.GroupName)
+	}
+	return nil
+}
+
+// isDirectory returns true if the passed path is a directory;
+// false otherwise.
+func isDirectory(path string) bool {
+	if d, err := os.Stat(path); err == nil {
+		if d.IsDir() {
+			return true
+		}
+	}
+	return false
+}
+
+// packageNamespace returns the namespace of the package
+// config files.
+func packageNamespace(packageDir string) (string, error) {
+	r := kio.LocalPackageReader{PackagePath: packageDir}
+	nodes, err := r.Read()
+	if err != nil {
+		return "", err
+	}
+	namespace := "default"
+	for _, node := range nodes {
+		rm, err := node.GetMeta()
+		if err != nil {
+			continue
+		}
+		if len(rm.ObjectMeta.Namespace) > 0 {
+			namespace = rm.ObjectMeta.Namespace
+		}
+		break
+	}
+	return namespace, nil
+}
+
+// defaultGroupName returns a string of the package name (directory)
+// with a random number suffix.
+func (i *InitOptions) defaultGroupName() string {
+	rand.Seed(i.Seed)
+	r := rand.Intn(1000000)
+	return fmt.Sprintf("%s-%06d", filepath.Base(i.Dir), r)
+}
+
+const groupNameRegexp = `^[a-zA-Z0-9-_\.]+$`
+
+// validateGroupName returns true of the passed group name is a
+// valid label value; false otherwise. The valid label values
+// are [a-z0-9A-Z] "-", "_", and "." The groupName must not
+// be empty, but it can not be more than 63 characters.
+func validateGroupName(groupName string) bool {
+	if len(groupName) == 0 || len(groupName) > 63 {
+		return false
+	}
+	re := regexp.MustCompile(groupNameRegexp)
+	return re.MatchString(groupName)
+}
+
+// fileExists returns true if a file at path already exists;
+// false otherwise.
+func fileExists(path string) bool {
+	f, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !f.IsDir()
+}
+
+func (i *InitOptions) fillInValues() string {
+	now := time.Now()
+	nowStr := now.Format("2006-01-02 15:04:05 MST")
+	manifestStr := configMapTemplate
+	manifestStr = strings.ReplaceAll(manifestStr, "<DATETIME>", nowStr)
```

Comment:
go has a templating engine in "text/template". We should use it.

Reply:
I looked into it, and it seemed like overkill. Let me add a TODO.